src/neo4j_genai/retrievers/external/pinecone/pinecone.py

```diff
@@ -77,7 +77,7 @@ class PineconeNeo4jRetriever(ExternalRetriever):
         id_property_neo4j (str): The name of the Neo4j node property that's used as the identifier for relating matches from Weaviate to Neo4j nodes.
         embedder (Optional[Embedder]): Embedder object to embed query text.
         return_properties (Optional[list[str]]): List of node properties to return.
-        result_formatter (Optional[Callable[[Any], Any]]): Function to transform a neo4j.Record to a RetrieverResultItem.
+        result_formatter (Optional[Callable[[neo4j.Record], RetrieverResultItem]]): Function to transform a neo4j.Record to a RetrieverResultItem.
```

Comment:
Future idea, not for this diff:

Assuming one of the goals of the library is allowing user not to write any Cypher (hence they can immediately start using Neo4J):

Should we provide sample 'result_formatter's?

Reply:
There is no need to write any Cypher in a `result_formatter`, just to access the returned objects. We also provide a default for when the parameter is null.